sequence_processing_pipeline/Pipeline.py

```diff
@@ -238,12 +254,21 @@ def generate_sample_information_files(self):
         Generate sample-information files in self.output_path.
         :return: A list of paths to sample-information-files.
         """
-        samples = []
-        for sample in self.sample_sheet.samples:
-            if sample['Sample_ID'].startswith('BLANK'):
-                samples.append((sample['Sample_ID'], sample['Sample_Project']))
+        if self.mapping_file is not None:
+            # TODO: Note we may need to introduce a conversion to sample_id
+            #  format from sample_name.
+            df = self.mapping_file[['sample_name', 'project_name']]
+            samples = list(df.to_records(index=False))
```

Comment:
Extra: you could do:
```
samples = [s for s in df.sample_name if 'BLANK' not in s] # or the startswith
projects = df.project_name.unique()
```

Reply:
samples needs to be a list of (sample-name, project-name) tuples. Will look into it though ty.